napari_allencell_annotator/controller/main_controller.py

```diff
@@ -73,20 +103,77 @@ def _csv_import_selected_evt(self, file_list: List[str]):
         file_list : List[str]
             The list containing one file name.
         """
-
+        self.already_annotated = None
         if file_list is None or len(file_list) < 1:
             self.images.view.alert("No selection provided")
         else:
-            # TODO
             file_path = file_list[0]
+            if os.path.splitext(file_path)[1] == ".json":
+                self.annots.read_json(file_path)
+
+            else:
+                proceed: bool = self.annots.view.popup("Would you like to use the images in this csv?")
+                file = open(file_path)
+
+                reader = csv.reader(file)
+                shuffled = next(reader)[1]
+                shuffled = self.str_to_bool(shuffled)
+                # annotation data header
+                annts = next(reader)[1]
+                # skip actual header
+                next(reader)
+                self.starting_row = None
+                if proceed:
+                    self.already_annotated = {}
+
+                    row_num: int = 0
+                    for row in reader:
+
+                        self.already_annotated[row[0]] = row[1::]
+                        if self.starting_row is None:
+                            for j in row[3::]:
+                                if j is None or j == "":
+                                    # if there is a none value for an annotation
+                                    self.starting_row = row_num
```

Comment:
could use a comment or two about what's going on here

Reply:
I have added comments here on my next PR where this method changes a lot!